planetscale/imports.go

```diff
@@ -0,0 +1,306 @@
+	"data_copy_pending":                DataImportCopyingData,
+	"data_copy_error":                  DataImportCopyingDataFailed,
+	"switch_traffic_workflow_pending":  DataImportSwitchTrafficPending,
+	"switch_traffic_workflow_running":  DataImportSwitchTrafficRunning,
+	"switch_traffic_workflow_error":    DataImportSwitchTrafficError,
+	"reverse_traffic_workflow_running": DataImportReverseTrafficRunning,
+	"reverse_traffic_workflow_error":   DataImportReverseTrafficError,
+	"cleanup_workflow_pending":         DataImportSwitchTrafficCompleted,
+	"cleanup_workflow_running":         DataImportDetachExternalDatabaseRunning,
+	"cleanup_workflow_error":           DataImportDetachExternalDatabaseError,
+	"ready":                            DataImportReady,
+}
+
+var importStateToDescMap = map[DataImportState]string{
+	DataImportPreparingDataCopy:             "Preparing to copy data from external database",
+	DataImportPreparingDataCopyFailed:       "Failed to copy data from external database",
+	DataImportCopyingData:                   "Copying data from external database",
+	DataImportCopyingDataFailed:             "Failed to copy data from external database",
+	DataImportSwitchTrafficPending:          "PlanetScale database is running in replica mode",
+	DataImportSwitchTrafficRunning:          "Switching PlanetScale database to primary mode",
+	DataImportSwitchTrafficError:            "Failed to switching PlanetScale database to primary mode",
+	DataImportReverseTrafficRunning:         "Switching PlanetScale database to replica mode",
+	DataImportReverseTrafficError:           "Failed to switching PlanetScale database to replica mode",
+	DataImportDetachExternalDatabaseRunning: "Detaching external database from  PlanetScale database",
+	DataImportDetachExternalDatabaseError:   "Failed to detach external database from  PlanetScale database",
+	DataImportReady:                         "Import has completed and your PlanetScale Database is now ready",
+}
+
+func (d DataImportState) String() string {
+	if val, ok := importStateToDescMap[d]; ok {
+		return val
+	}
+
+	panic("unknown data import state")
+}
+
+type DataImport struct {
+	ID                 string `json:"id"`
+	ImportState        DataImportState
+	State              string           `json:"state"`
+	Errors             string           `json:"import_check_errors"`
+	StartedAt          *time.Time       `json:"started_at"`
+	FinishedAt         *time.Time       `json:"finished_at"`
+	DeletedAt          *time.Time       `json:"deleted_at"`
+	ExternalDataSource DataImportSource `json:"data_source"`
+}
+
+func (di *DataImport) ParseState() {
+	if val, ok := stateToImportStateMap[di.State]; ok {
+		di.ImportState = val
+		return
+	}
+
+	panic("unknown data import state " + di.State)
+}
+
+type TestDataImportSourceRequest struct {
+	Organization string           `json:"organization"`
+	Database     string           `json:"database_name"`
+	Source       DataImportSource `json:"connection"`
```

Comment:
JSON tag is wrong or the variable name should be renamed to `Connection`. Wdty?

Reply:
Renamed the property to `Connection`